File: output/android/page_viewed.kt
enum class PageViewedEventPagename(val name: String) {
      APPLICATION_SIGN_IN("application.sign_in"),
      APPLICATION_SETTINGS("application.settings"),
      SETTINGS_NOTIFICATIONS("settings.notifications"),
    
}

data class PageViewedEvent(val pageName: PageViewedEventPagename) {
    val name: String = "page_viewed"
    val parameters: [String: Any]? {
        return [
            "page_name": self.pageName.name
        ]
    }
}

File: output/ios/page_viewed.swift
struct PageViewedEvent {
    enum Name: String {
      case applicationSignIn = "application.sign_in"
      case applicationSettings = "application.settings"
      case settingsNotifications = "settings.notifications"
    
    }
    
    var name: String {
        return "page_viewed"
    }
    
    var parameters: [String: Any]? {
        return [
            "page_name": self.pageName.rawValue
        ]
    }
    
    private let pageName: Name
    init(pageName: Name) {
        self.pageName = pageName
    }
}
